src/api/v4/dataview/base.js

```diff
@@ -0,0 +1,227 @@
+};
+
+/**
+ * Enable the dataview
+ *
+ * @return {carto.dataview.Base} this
+ * @api
+ */
+Base.prototype.enable = function () {
+  return this._setEnabled(true);
+};
+
+/**
+ * Disable the dataview
+ *
+ * @return {carto.dataview.Base} this
+ * @api
+ */
+Base.prototype.disable = function () {
+  return this._setEnabled(false);
+};
+
+/**
+ * Return true if the dataview is enabled
+ *
+ * @return {boolean}
+ * @api
+ */
+Base.prototype.isEnabled = function () {
+  return this._enabled;
+};
+
+/**
+ * Set the dataview column
+ *
+ * @param  {string} column
+ * @return {carto.dataview.Base} this
+ * @api
+ */
+Base.prototype.setColumn = function (column) {
+  this._checkColumn(column);
+  this._column = column;
+  if (this._internalModel) {
+    this._internalModel.set('column', this._column);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview column
+ *
+ * @return {string} Current dataview column
+ * @api
+ */
+Base.prototype.getColumn = function () {
+  return this._column;
+};
+
+Base.prototype.getData = function () {
+  throw new Error('getData must be implemented by the particular dataview.');
+};
+
+// syncOnBoundingChanges
+// isSyncedOnBoundingChanges
+
+// Protected methods
+
+Base.prototype._initialize = function (source, column, options) {
+  options = this._defaultOptions(options);
+
+  this._checkSource(source);
+  this._checkColumn(column);
+  this._checkOptions(options);
+
+  this._source = source;
+  this._column = column;
+  this._options = options;
+  this._status = STATUS.NOT_LOADED;
+  this._enabled = true;
+};
+
+Base.prototype._defaultOptions = function (options) {
+  throw new Error('_defaultOptions must be implemented by the particular dataview.');
+};
+
+Base.prototype._checkSource = function (source) {
+  if (!(source instanceof SourceBase)) {
+    throw new TypeError('Source property is required.');
+  }
+};
+
+Base.prototype._checkColumn = function (column) {
+  if (_.isUndefined(column)) {
+    throw new TypeError('Column property is required.');
+  }
+  if (!_.isString(column)) {
+    throw new TypeError('Column property must be a string.');
+  }
+  if (_.isEmpty(column)) {
+    throw new TypeError('Column property must be not empty.');
+  }
+};
+
+Base.prototype._checkOptions = function (options) {
+  throw new Error('_checkOptions must be implemented by the particular dataview.');
+};
+
+Base.prototype._createInternalModel = function (engine) {
+  throw new Error('_createInternalModel must be implemented by the particular dataview.');
+};
+
+Base.prototype._setEnabled = function (enabled) {
+  this._enabled = enabled;
+  if (this._internalModel) {
+    this._internalModel.set('enabled', enabled);
+  }
+  return this;
+};
+
+Base.prototype._listenToInternalModelEvents = function () {
+  if (this._internalModel) {
+    this.listenTo(this._internalModel, 'change:data', this._onDataChanged);
+    this.listenTo(this._internalModel, 'change:column', this._onColumnChanged);
```

Comment:
In which ways can be the column changed?

By the user through the public class and ... ?

Reply:
Only with dataviewInstance.setColumn('...'), but we are adding events for every property that can be set